src/main/java/com/hillel/items_exchange/service/UserService.java

```diff
@@ -1,23 +1,59 @@
 package com.hillel.items_exchange.service;
 
-import com.hillel.items_exchange.dto.UserRegistrationDto;
-import com.hillel.items_exchange.model.Role;
-import com.hillel.items_exchange.model.User;
-import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
-import org.springframework.stereotype.Service;
-
 import java.util.Optional;
 
-@Service
-public interface UserService {
-
-    boolean existsByUsernameOrEmail(String usernameOrEmail);
-
-    boolean existsByUsernameOrEmailAndPassword(String usernameOrEmail, String password, BCryptPasswordEncoder bCryptPasswordEncoder);
+import org.springframework.context.annotation.Bean;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.stereotype.Service;
 
-    Optional<User> findByUsernameOrEmail(String usernameOrEmail);
+import lombok.RequiredArgsConstructor;
 
-    boolean save(User user);
+import com.hillel.items_exchange.dao.UserRepository;
+import com.hillel.items_exchange.dto.UserRegistrationDto;
+import com.hillel.items_exchange.mapper.UserMapper;
+import com.hillel.items_exchange.model.Role;
+import com.hillel.items_exchange.model.User;
 
-    void registerNewUser(UserRegistrationDto userRegistrationDto, BCryptPasswordEncoder bCryptPasswordEncoder, Role role);
+@Service
+@RequiredArgsConstructor
+public class UserService {
+
+    private final UserRepository userRepository;
+
+    @Bean
+    public BCryptPasswordEncoder bCryptPasswordEncoder() {
+        return new BCryptPasswordEncoder();
+    }
```

Comment:
According to Spring reference doc: "Only using @Bean methods within @Configuration classes is a recommended approach of ensuring that 'full' mode is always used. This will prevent the same @Bean method from accidentally being invoked multiple times and helps to reduce subtle bugs that can be hard to track down when operating in 'lite' mode."
Maybe it should be moved to AppConfig class?

Reply:
I agree with youu